### ISSUE TITLE
Allow  NULL in x-db-type jsonb 

### ISSUE THREAD

**Issue by SOHELAHMED7** [cebe/yii2-openapi#89]
At the moment if I create a property named (for e.g.) `res_header`  with required `false` and x-db-type `jsonb`, migrations generated have `'response_headers' => 'jsonb NOT NULL',`

Example: Del.yaml

```yaml
# del:
title: Del
x-table: del
type: object


required:
  - id 

  # ...
  
  # - res_header #not required
  

  # ...
  

properties:

  id:
    type: integer
    readOnly: true
    example: 1337

  res_header:
    type: array
    x-db-type: jsonb
    example:
      - {"atl-traceid": "16eae2fc924", "content-encoding": "gzip", "content-type": "text/html", "date": "Thu, 02 Dec 2021 12:59:38 GMT", "referrer-policy": "strict-origin-when-cross-origin"}
    description: Response header 
    items:
      type: object  

    # ...
      
```

Expected:

Yii2 migration should have allow null

e.g. `'response_headers' => 'jsonb',`

**Comment by Insolita** [comment]
As I remember it is not a bug, it is a feature - for json columns allowNull force disabled
https://github.com/cebe/yii2-openapi/blob/master/src/lib/items/Attribute.php#L247-L248

but I don't remember why, @cebe ?

**Comment by cebe** [comment]
yes, when having a json field I see no use case where NULL makes sense to have. I do not remember the concrete case on which we decided to make it this way but it is explicitly disallowed to avoid extra NULL checks.

**Comment by SOHELAHMED7** [comment]
My use case is like:

I have webhook delivery table with column `response_headers` (jsonb allow null)

I need to put Delivery ID (PK) in request header

So I first save Delivery (not yet sent/HTTP call)

So `response_header` is saved as `null` in DB

I get Delivery ID, I put Delivery ID in Request Header, I send HTTP request. 

I get response headers now and store it `response_headers` column

---

So in conclusion I should proceed with `{}` or `[]` instead of `null` and this behaviour will remain as it is, right?

**Comment by cebe** [comment]
so your use case is:

- `response_headers = NULL` = headers not received/sent yet
- `response_headers = []` no headers sent.

that looks like a valid case, but there is another case to consider:

`null` is a valid JSON value, so what is the difference between `response_headers = JSON "null"` and `response_headers = SQL "null"`.

```php
var_dump(json_decode('null'));
// output: NULL
```

I do not think there is avalid case for allowing SQL NULL here.

**Comment by SOHELAHMED7** [comment]
Agree mostly as far as we use Yii's ActiveRecord `$model` for saving data. Since Yii [automatically cast](https://github.com/yiisoft/yii2/blob/master/framework/db/JsonExpression.php#L18) array/null to json for json field, `'null'` will be stored in DB and that is indeed ambiguous. 

E.g. :

```
$model = new Post // (yii\db\ActiveRecord);
$model->title = 'title';
// $model->meta = ['gtag-id' => '465'];  # not null
// or 
// $model->meta = null; # JSON `'null'`, if meta column have allow null in DB
$model->save();
```

Here if meta is null `'null'` (as string JSON null) will be saved and it is ambiguous.

But if we need to store record without using Active Record; say

```
Yii::$app->db->createCommand()->insert('post', [
                'title' => 'title', 
            ])->execute();
```

here meta will be `null` automatically  (of DB and not of JSON `'json'` (as string JSON null) )

**Comment by cebe** [comment]
good point, but as always when working without abstraction like AR you have to do a bit more work to cover all cases.
